src/main/java/org/trigon/hardware/misc/leds/AddressableLEDStrip.java

```diff
@@ -0,0 +1,182 @@
+            if (!shouldLoop) {
+                getDefaultCommand().schedule();
+                return;
+            }
+            amountOfColorFlowLEDs = inverted ? numberOfLEDs : 0;
+        }
+        if (inverted) {
+            setLEDColors(color, numberOfLEDs - amountOfColorFlowLEDs, numberOfLEDs - 1);
+            return;
+        }
+        setLEDColors(color, 0, amountOfColorFlowLEDs);
+    }
+
+    @Override
+    void alternateColor(Color firstColor, Color secondColor, double intervalSeconds) {
+        double currentTime = Timer.getFPGATimestamp();
+        if (currentTime - lastLEDMovementTime > intervalSeconds) {
+            alternateColor = !alternateColor;
+            lastLEDMovementTime = currentTime;
+        }
+        if (alternateColor) {
+            for (int i = 0; i < numberOfLEDs; i++)
+                LED_BUFFER.setLED(i + indexOffset, i % 2 == 0 ? firstColor : secondColor);
+            return;
+        }
+        for (int i = 0; i < numberOfLEDs; i++)
+            LED_BUFFER.setLED(i + indexOffset, i % 2 == 0 ? secondColor : firstColor);
+    }
+
+
+    @Override
+    void rainbow(double brightness, double speed) {
+        for (int led = 0; led < numberOfLEDs; led++) {
+            final int hue = (int) (rainbowFirstPixelHue + (led * 180 / numberOfLEDs) % 180);
+            LED_BUFFER.setHSV(led + indexOffset, hue, 255, 128);
+        }
+        if (inverted) {
+            rainbowFirstPixelHue -= 3;
+            if (rainbowFirstPixelHue < 0)
+                rainbowFirstPixelHue += 180;
+            return;
+        }
+        rainbowFirstPixelHue += 3;
+        rainbowFirstPixelHue %= 180;
+    }
+
+    @Override
+    void sectionColor(int amountOfSections, Supplier<Color>[] colors) {
+        if (amountOfSections != colors.length)
+            throw new IllegalArgumentException("Amount of sections must be equal to the amount of colors");
+        final int LEDSPerSection = (int) Math.floor(numberOfLEDs / amountOfSections);
+        setSectionColor(amountOfSections, LEDSPerSection, colors);
+    }
+
+    @Override
+    void resetLEDSettings() {
+        lastBreatheLED = indexOffset;
+        lastLEDMovementTime = Timer.getFPGATimestamp();
+        rainbowFirstPixelHue = 0;
+        areLEDsOnForBlinking = false;
+        alternateColor = true;
+        amountOfColorFlowLEDs = 0;
+    }
+
+    private void setSectionColor(int amountOfSections, int LEDSPerSection, Supplier<Color>[] colors) {
+        if (inverted) {
+            for (int i = 0; i < amountOfSections; i++)
+                setLEDColors(colors[amountOfSections - i - 1].get(), LEDSPerSection * i, i == amountOfSections - 1 ? numberOfLEDs - 1 : LEDSPerSection * (i + 1) - 1);
+            return;
+        }
+        for (int i = 0; i < amountOfSections; i++)
+            setLEDColors(colors[i].get(), LEDSPerSection * i, i == amountOfSections - 1 ? numberOfLEDs - 1 : LEDSPerSection * (i + 1) - 1);
```

Comment:
Compress this into one line. Use the ternary operator

Reply:
No, it's really ugly and hard to understand with a tenary operator. Just because something can be written in one line doesn't mean it should.